ecommerce/extensions/iap/api/v1/views.py

```diff
@@ -0,0 +1,95 @@
+# pylint: disable=no-else-return
+
+
+import logging
+import time
+from django.http import JsonResponse
+from django.utils.html import escape
+from django.utils.translation import ugettext as _
+from edx_rest_framework_extensions.permissions import LoginRedirectIfUnauthenticated
+from oscar.apps.basket.views import *  # pylint: disable=wildcard-import, unused-wildcard-import
+from rest_framework.views import APIView
+from ecommerce.extensions.analytics.utils import track_segment_event
+from ecommerce.extensions.basket.constants import EMAIL_OPT_IN_ATTRIBUTE
+from ecommerce.extensions.basket.exceptions import BadRequestException, RedirectException
+from ecommerce.extensions.basket.utils import prepare_basket
+from ecommerce.extensions.basket.views import BasketLogicMixin
+from ecommerce.extensions.order.exceptions import AlreadyPlacedOrderException
+from ecommerce.extensions.partner.shortcuts import get_partner_for_site
+
+BasketAttribute = get_model('basket', 'BasketAttribute')
+BasketAttributeType = get_model('basket', 'BasketAttributeType')
+logger = logging.getLogger(__name__)
+Product = get_model('catalogue', 'Product')
+Voucher = get_model('voucher', 'Voucher')
+
+
+class MobileBasketAddItemsView(BasketLogicMixin, APIView):
```

Comment:
If this code (or any other) is copied from elsewhere, please add a comment directing readers to the original code elsewhere in the repo.

Reply:
Reference added.